zerver/tests/test_messages.py

```diff
@@ -586,6 +590,82 @@ def test_applying_markdown(self):
         self.assertEqual(message.rendered_content, expected_content)
         self.assertEqual(message.rendered_content_version, bugdown.version)
 
+    def test_reaction(self):
+        # type: () -> None
+        sender = get_user_profile_by_email('othello@zulip.com')
+        receiver = get_user_profile_by_email('hamlet@zulip.com')
+        recipient = Recipient.objects.get(type_id=receiver.id, type=Recipient.PERSONAL)
+        sending_client = make_client(name="test suite")
+        message = Message(
+            sender=sender,
+            recipient=recipient,
+            subject='whatever',
+            content='hello **world**',
+            pub_date=timezone.now(),
+            sending_client=sending_client,
+            last_edit_time=timezone.now(),
+            edit_history='[]'
+        )
+        message.save()
+
+        reaction = Reaction.objects.create(
+            message=message, user_profile=sender,
+            emoji_name='simple_smile')
+        row = Message.get_raw_db_rows([message.id])[0]
+        msg_dict = MessageDict.build_dict_from_raw_db_row(row,
+               apply_markdown=True)
+        self.assertEqual(msg_dict['reactions'][0]['emoji_name'],
+                         reaction.emoji_name)
+        self.assertEqual(msg_dict['reactions'][0]['user']['id'],
+                         sender.id)
+        self.assertEqual(msg_dict['reactions'][0]['user']['email'],
+                         sender.email)
+        self.assertEqual(msg_dict['reactions'][0]['user']['full_name'],
+                         sender.full_name)
+
+
+class SewMessageAndReactionTest(ZulipTestCase):
+    def test_sew_messages_and_reaction(self):
+        # type: () -> None
+        sender = get_user_profile_by_email('othello@zulip.com')
+        receiver = get_user_profile_by_email('hamlet@zulip.com')
+        pm_recipient = Recipient.objects.get(type_id=receiver.id, type=Recipient.PERSONAL)
+        stream_name = u'Çiğdem'
+        stream = self.make_stream(stream_name)
+        stream_recipient = Recipient.objects.get(type_id=stream.id, type=Recipient.STREAM)
+        sending_client = make_client(name="test suite")
+
+        needed_ids = []
+        for i in range(100):
```

Comment:
This is a pretty expensive test (doing 200 database queries); can we do a much smaller number (e.g. 5) and still get the same testing benefit?